Test/TestFramework/Shared/StubPlatform.cs

```diff
@@ -25,7 +27,18 @@ public IDebugOutput GetDebugOutput()
 
         public string GetEnvironmentVariable(string name)
         {
-            return Environment.GetEnvironmentVariable(name);
+            try
+            {
+                return Environment.GetEnvironmentVariable(name);
+            }
+            catch (SecurityException e)
+            {
+                // Not being able to get the environment variable is crucial and
+                // cant be dodged without some sort of default values.
+                // Since this is not an option in this context, lets just log and rethrow.
+                CoreEventSource.Log.FailedToLoadEnvironmentVariables(e.ToString());
+                throw e;
```

Comment:
Do not re-throw an exception like this. This will destroy the call stack. [Click here for more information](https://docs.microsoft.com/en-us/cpp/cpp/try-throw-and-catch-statements-cpp?view=vs-2017#remarks): 
```suggestion
                throw;
```

Reply:
Note that [GetEnvironmentVariable ](https://docs.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariables?view=netframework-4.7.2)can throw other exceptions than SecurityException.

I think it would be appropriate to catch all exceptions here `catch (Exception e)`